.semaphore/semaphore.yml

```diff
@@ -1,26 +1,67 @@
 version: v1.0
-name: Test MyApp
+name: Rails Base
 agent:
   machine:
     type: e1-standard-2
     os_image: ubuntu1804
-
+execution_time_limit:
+  hours: 1
+auto_cancel:
+  queued:
+    when: 'true'
+fail_fast:
+  stop:
+    when: 'true'
+global_job_config:
+  env_vars:
+    - name: RAILS_ENV
+      value: test
+    - name: DATABASE_URL
+      value: 'postgresql://postgres@localhost/test?encoding=utf8'
+    - name: BUNDLE_ARGS
+      value: '--deployment --without development staging production --jobs 4'
+  prologue:
+    commands:
+      - checkout
+      - cache restore
 blocks:
-  - name: Test
+  - name: Setup
+    task:
+      jobs:
+        - name: Bundle
+          commands:
+            - 'bundle install ${BUNDLE_ARGS}'
+            - cache store
+      secrets:
+        - name: preinternship
+  - name: Quality
     task:
-      env_vars:
-        - name: DATABASE_URL
-          value: "postgresql://postgres:@0.0.0.0:5432/myapp_test"
-        - name: RAILS_ENV
-          value: test
       jobs:
-      - name: Test
+        - name: Quality
+          commands:
+            - 'bundle install ${BUNDLE_ARGS} --local'
```

Comment:
please, see documentation about cachind https://docs.semaphoreci.com/essentials/caching-dependencies-and-directories/